Task: Spring Boot Portfolio Project — Part 2(DTO). Executor: Oleksii Bashkirov.

OleksiiBashkirov opened the pull request:
Improved project structure and functionality for online-book-store.
Refactored packages:
- config
- controller
- dto
- exception
- mapper
- model
- repository
- service

Edited: 
- BookService interface: updated methods save, findById and findAll.
- BookServiceImpl class: implemented methods for the updated interface.
- BookRepository interface: added Optional return type to the findById method.
- BookRepositoryImpl class: updated methods using jakarta.persistence library(EntityManagerFactory, EntityManager, EntityTransaction).
- Application class: removed CommandLineRunner bean.
- pom.xml file: added dependencies for mapstruct, mapstruct-processor,lombok-mapstruct-binding,mysql-connector-j and lombok. Configured maven-compiler-plugin.

Created:
- BookController class with @RestController, @RequestMapping(value = "/api/books"), and methods getAll, getBookById and createBook.
- BookDto class with fields: id, title, author, isbn, price, description and coverImage.
- CreateBookRequestDto class with the same fields.
- BookMapper interface with @Mapper(config = MapperConfig.class) annotation.
- EntityNotFoundException class.
- MapperConfig Class with @org.mapconstruct.MapperConfig annotation.

OleksiiBashkirov committed:
commit  the online-book-store to the repository;

OleksiiBashkirov committed:
Update README.md

OleksiiBashkirov committed:
fixed checkstyle errors;

OleksiiBashkirov committed:
Merge branch 'main' of https://github.com/OleksiiBashkirov/springBoot_repo

OleksiiBashkirov committed:
Add Book entity, repository, and service

OleksiiBashkirov committed:
added empty line to the end of the file;

OleksiiBashkirov committed:
removed the redundant lines with the text "comment";

OleksiiBashkirov committed:
- Added new lines to the Book class

OleksiiBashkirov committed:
"Update .gitignore and remove .mvn directory from repository"

OleksiiBashkirov committed:
Improved project structure and functionality for online-book-store.
Refactored packages:
- config
- controller
- dto
- exception
- mapper
- model
- repository
- service

Edited:
- BookService interface: updated methods save, findById and findAll.
- BookServiceImpl class: implemented methods for the updated interface.
- BookRepository interface: added Optional return type to the findById method.
- BookRepositoryImpl class: updated methods using jakarta.persistence library(EntityManagerFactory, EntityManager, EntityTransaction).
- Application class: removed CommandLineRunner bean.
- pom.xml file: added dependencies for mapstruct, mapstruct-processor,lombok-mapstruct-binding,mysql-connector-j and lombok. Configured maven-compiler-plugin.

Created:
- BookController class with @RestController, @RequestMapping(value = "/api/books"), and methods getAll, getBookById and createBook.
- BookDto class with fields: id, title, author, isbn, price, description and coverImage.
- CreateBookRequestDto class with the same fields.
- BookMapper interface with @Mapper(config = MapperConfig.class) annotation.
- EntityNotFoundException class.
- MapperConfig Class with @org.mapconstruct.MapperConfig annotation.

Elena-Bruyako commented on src/main/java/mate/academy/controller/BookController.java:
```diff
@@ -0,0 +1,35 @@
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/books")
```
```suggestion
@RequestMapping("/books")
```

add `server.servlet.context-path=/api` to application.properties

Elena-Bruyako commented on src/main/java/mate/academy/model/Book.java:
```diff
@@ -0,0 +1,42 @@
+
+@Entity
+@Table(name = "books")
+@Data
```
Remove 
```
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
```

Add
```
@Getter
@Setter
```

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,41 @@
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .collect(Collectors.toList());
```
```suggestion
                .toList();
```

Elena-Bruyako reviewed (changes requested):
see pr

OleksiiBashkirov committed:
BookController class:
- changed @RequestMapping value to "/books".
- added server.servlet.context-path=/api to application.properties.

Book class:
- removed unnecessary annotations(@Data,@NoArgsConstructor,@AllArgsConstructor,@Builder
- added @Getter and @Setter annotations.

BookServiceImpl class:
- replaced collect(Collectors.toList()) with .toList().

OleksiiBashkirov committed:
Merge branch 'main' into book-entity

OleksiiBashkirov committed:
Update BookRepository.java

removed extra line
